### ISSUE TITLE
Android combining character input (using MessagEase) doesn't work correctly

### ISSUE THREAD

**Issue by LaPingvino** [fyne-io/fyne#2066]
### Describe the bug:
When I type words which use combining characters on my MessagEase keyboard (e.g. ĝojo) it looks like it would normally remove the elements and put the final character instead, but the removal doesn't work with my test fyne app (i.e. it comes out as g^ĝojo).

### To Reproduce:
Steps to reproduce the behaviour:
1. Install MessagEase keyboard
2. Try a Fyne app that uses Entry
3. Type g, then swipe up from the top middle key to type ^, which should normally combine
4. See above result instead (g^ĝ)

### Device (please complete the following information):
 - **OS:** Android
 - **Version:** 10
 - **Go version:** fyne-cross
 - **Fyne version:** 2.0

**Comment by andydotxyz** [comment]
This is not a duplicate of, but could relate somehow to, #73

**Comment by LaPingvino** [comment]
MessagEase confirms they use selection then typing over it to implement the combining characters: https://twitter.com/MessagEase/status/1369797409539194880

**Comment by andydotxyz** [comment]
OK great thanks.
This is challenging - in essence we will have to simulate backspace and re-type when we detect a text change inexpected in the native entry.

**Comment by andydotxyz** [comment]
Can you please test with `v2.1.0-rc1` @LaPingvino because we re-worked the Android keyboard code so it could have resolved this issue...

**Comment by LaPingvino** [comment]
Could you add an APK to test with? No quick access to recompile my test app.

**Comment by andydotxyz** [comment]
I didn't know what architecture your test device was so I have created a fyne-demo APK that combines all architectures without debug symbols stripped, so it is quite large. However if you head to the Input or Form items in the Widget area you should be able to test to see if keyboard is working better.

https://www.dropbox.com/s/imvsxmrkr1vu57y/fyne_demo-2.1.0-rc1.apk?dl=0

**Comment by andydotxyz** [comment]
> just like my Chromebook (where I have issues compiling Fyne in the built-in Linux for the Android target)

I don't think that the Android NDK supports building like that - they don't ship an arm64 build toolkit.

**Comment by andydotxyz** [comment]
> Before I typed g^ and it became g^ĝ
> now I type g^ and it becomes gĝ
> I expect it to become just ĝ

I am sorry we have not completely solved this. I will look into it more after release, the code is easier to understand now so it may be easier to fix than before :)
